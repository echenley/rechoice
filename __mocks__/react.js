const React = require.requireActual('react')

React.Fragment = 'Fragment'

module.exports = React
