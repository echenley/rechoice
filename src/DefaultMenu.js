// @flow

import * as React from 'react'
import type { MenuRenderer } from './types'

const Menu: MenuRenderer = ({
  children,
  className,
  isOpen,
  listProps,
  loadingIndicator,
  menuRef,
  state,
}) => {
  if (!isOpen) {
    return null
  }

  let content = null

  if (state === 'loading') {
    const loader = loadingIndicator || 'Loading resultsâ€¦'
    content = <div className="loading">{loader}</div>
  } else if (state === 'error') {
    content = <div className="error">{'Something went wrong.'}</div>
  } else if (React.Children.count(children) === 0) {
    content = <div className="empty">{'No results.'}</div>
  } else {
    content = <ul {...listProps}>{children}</ul>
  }

  return (
    <div className={className} ref={menuRef}>
      {content}
    </div>
  )
}

export default Menu
