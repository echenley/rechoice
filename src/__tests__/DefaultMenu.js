import * as React from 'react'
import { shallow } from 'enzyme'
import DefaultMenu from '../DefaultMenu'

const children = [
  <div key="1">{'Child 1'}</div>,
  <div key="2">{'Child 2'}</div>,
  <div key="3">{'Child 3'}</div>,
]

const renderComponent = props => {
  return shallow(
    <DefaultMenu
      // eslint-disable-next-line react/no-children-prop
      children={children}
      className="className"
      isOpen={true}
      listProps={{ listProp1: 'listProp1' }}
      setMenuRef={() => {}}
      state="success"
      {...props}
    />
  )
}

describe('DefaultMenu', () => {
  it('renders correctly when loading', () => {
    expect(renderComponent({ state: 'loading' }).getElement()).toMatchSnapshot()
  })

  it('renders loadingIndicator if available', () => {
    expect(
      renderComponent({
        state: 'loading',
        loadingIndicator: <div className="loading-indicator" />,
      }).getElement()
    ).toMatchSnapshot()
  })

  it('renders correctly for an error', () => {
    expect(renderComponent({ state: 'error' }).getElement()).toMatchSnapshot()
  })

  it('renders correctly with no results', () => {
    expect(renderComponent({ children: [] }).getElement()).toMatchSnapshot()
  })

  it('renders correctly with results', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('renders correctly when isOpen is false (null)', () => {
    expect(renderComponent({ isOpen: false }).getElement()).toMatchSnapshot()
  })
})
