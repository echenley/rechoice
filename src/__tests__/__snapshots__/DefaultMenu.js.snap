// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultMenu renders correctly for an error 1`] = `
<div
  className="className"
>
  <div
    className="error"
  >
    Something went wrong.
  </div>
</div>
`;

exports[`DefaultMenu renders correctly when isOpen is false (null) 1`] = `null`;

exports[`DefaultMenu renders correctly when loading 1`] = `
<div
  className="className"
>
  <div
    className="loading"
  >
    Loading resultsâ€¦
  </div>
</div>
`;

exports[`DefaultMenu renders correctly with no results 1`] = `
<div
  className="className"
>
  <div
    className="empty"
  >
    No results.
  </div>
</div>
`;

exports[`DefaultMenu renders correctly with results 1`] = `
<div
  className="className"
>
  <ul
    listProp1="listProp1"
  >
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
    <div>
      Child 3
    </div>
  </ul>
</div>
`;

exports[`DefaultMenu renders loadingIndicator if available 1`] = `
<div
  className="className"
>
  <div
    className="loading"
  >
    <div
      className="loading-indicator"
    />
  </div>
</div>
`;
